### PR TITLE
Cloud NAT Rules and Configurable TCP Time Wait

### PR DESCRIPTION
**Pull request by DrFaust92**
Closes https://github.com/hashicorp/terraform-provider-google/issues/10666

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: add `rules` and `tcp_time_wait_timeout_sec` to `google_compute_router_nat`
```

### CODE REVIEW

**Comment by DrFaust92** [comment]
added test + fixes to schema

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -13545,6 +13550,56 @@ objects:
               - :ERRORS_ONLY
               - :TRANSLATIONS_ONLY
               - :ALL
+      - !ruby/object:Api::Type::Array
+        name: rules
```
Comment (shuyama1) [inline mmv1/products/compute/api.yaml:13554]: Are these fields updatable? Can we also add an update test for this feature?

**Review by shuyama1 — COMMENTED** [review]
Thanks for making the changes! Sorry for the delay on reviewing this.

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (DrFaust92) [reply mmv1/products/compute/api.yaml:13575]: added

File: mmv1/products/compute/api.yaml [hunk 2]
```diff
@@ -13525,6 +13525,11 @@ objects:
           Timeout (in seconds) for TCP transitory connections.
           Defaults to 30s if not set.
         default_value: 30
+      - !ruby/object:Api::Type::Integer
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply mmv1/products/compute/api.yaml:13549]: added

**Comment by DrFaust92** [comment]
Thanks for the comments [shuyama1](https://github.com/shuyama1), ive added relevant tests.

**Review by shuyama1 — APPROVED** [review]
LGTM! Thank you!

**Comment by shuyama1** [comment]
Test starts to fail in our nightly run due to:
```
provider_test.go:309: Step 3/4 error: After applying this test step, the plan was not empty.
stdout:
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
~ update in-place
Terraform will perform the following actions:
# google_compute_router_nat.foobar will be updated in-place
~ resource "google_compute_router_nat" "foobar" {
id                                  = "ci-test-project-188019/us-central1/tf-test-router-nat-kal6esk7c1/tf-test-router-nat-kal6esk7c1"
name                                = "tf-test-router-nat-kal6esk7c1"
# (16 unchanged attributes hidden)
~ rules {
~ match       = "inIpRange(destination.ip, '3.3.0.0/16') || inIpRange(destination.ip, '4.4.0.0/16')" -> "inIpRange(destination.ip, '1.1.0.0/16') || inIpRange(destination.ip, '5.5.0.0/16')"
~ rule_number = 2 -> 1
~ action {
~ source_nat_active_ips = [
- "https://www.googleapis.com/compute/v1/projects/ci-test-project-188019/regions/us-central1/addresses/tf-test-router-nat-kal6esk7c1-addr-3",
+ "projects/ci-test-project-188019/regions/us-central1/addresses/tf-test-router-nat-kal6esk7c1-addr-2",
]
# (1 unchanged attribute hidden)
}
}
~ rules {
~ match       = "inIpRange(destination.ip, '1.1.0.0/16') || inIpRange(destination.ip, '5.5.0.0/16')" -> "inIpRange(destination.ip, '3.3.0.0/16') || inIpRange(destination.ip, '4.4.0.0/16')"
~ rule_number = 1 -> 2
~ action {
~ source_nat_active_ips = [
- "https://www.googleapis.com/compute/v1/projects/ci-test-project-188019/regions/us-central1/addresses/tf-test-router-nat-kal6esk7c1-addr-2",
+ "projects/ci-test-project-188019/regions/us-central1/addresses/tf-test-router-nat-kal6esk7c1-addr-3",
]
+ source_nat_drain_ips  = []
}
}
}
Plan: 0 to add, 1 to change, 0 to destroy.
```

Reverting this change for now.

**Pull request merged**